cms/cms-import.md

```diff
@@ -0,0 +1,275 @@
+|0002-20180910|Let's go drinking this Friday! |Today is such a tired day|
+|0003-20180911|Of course! |Skygear is so awesome|
+
+Configurable items:
+
+- <a href='./#record_type'>`record_type`</a>
+- <a href='./#identifier'>`identifier`</a>
+- <a href='./#handle_duplicated_identifier'>`handle_duplicated_identifier`</a>
+- <a href='./#fields'>`fields`</a>
+
+<a name='record_type'></a>
+
+#### `record_type`
+The database table this import is linked to.
+```yml
+imports:
+  import-name-random: 
+    record_type: 
+```
+
+<a name='identifier'></a>
+
+#### `identifier`
+The field that Skygear uses to identify a record. If `identifier` is not provided, Skygear will use `_id` to identify records. 
+
+If a record in the database is found to have the same identifier as the imported record, Skygear will update the record in the database.
+
+```yml
+imports:
+  import-name-random: 
+    identifier: 
+```
+<a name='handle_duplicated_identifier'></a>
+
+#### `handle_duplicated_identifier`
+The way to handle duplicated records when If more than 1 record in the database is found to have the same identifier as the imported record.
+
+```yml
+imports:
+  import-name-random: 
+    handle_duplicated_identifier: throw-error or use-first
+```
+
+- `throw-error` will return an error about this record. If you want to reject the whole import when there are errors, you can set `atomic: true` in the `action` setting (will be explained below).
+
+- `use-first` will randomly accept one of the duplicated records.
+
+<a name='fields'></a>
+
+#### `fields`
+Fields the users should fill in. 
+
+For relational records, Import support only **reference** (i.e. 1-many).
+
+```yml
+imports:
+  import-name-random: 
+    fields:
+      - name: a normal field
+      - name: a reference field
+        reference_target: the referenced table 
+        reference_field_name: the field user should fill in the CSV to idenitfy the referenced record
+        handle_duplicated_reference: throw-error or use-first
+```
+- `throw-error` will return an error. If you want to reject the whole import when there are errors, you can set `atomic: true` in the `action` setting (will be explained below).
+
+- `use-first` will randomly assign the record to one of the duplicated references.
+
+### Add the import button
+
+To add an import button to the page, you need configure it in the list view. 
+
+Example usage:
+
+Suppose now we are adding an **import** button on the `comment`'s list view.
+
+```yml
+site:
+  - type: Record
+    name: comment
+
+records:
+  comment:
+    record_type: comment
+    list:
+      actions:
+        - type: Import
+          name: import-comment # this is defined in the import section
+          label: Import
+          atomic: true 
+```
+
+- `atomic`: When `atomtic` is set to true, the whole import will be rejected if there are errors from `handle_duplicated_reference` and `handle_duplicated_identifier`. Default is false.
```

Comment:
Default is `undefined`, but not false.

`undefined` means user can choose one of them in the import dialog.

Reply:
BTW, `undefined` means not setting any value in the config.

Reply:
> undefined means user can choose one of them in the import dialog.

To clarify, do you mean there will be a pop up for user to choose how to handle error at the time they press the import button?

Reply:
Yes, something like this.

<img width="1792" alt="screen shot 2018-07-17 at 5 35 01 pm" src="https://user-images.githubusercontent.com/12228201/42809252-c2612e84-89e7-11e8-9d7d-527cd419ee14.png">

Reply:
Thanks. Fixed.